### PR TITLE
Clean up judgehost diskspace by default.

### LINKED ISSUE #943: Judgehosts should (optionally) clean up old judging data
For long running contests(or even short ones where you don't have a ton of disk space) it'd be nice for judgehosts to have an option to clean up old judging data, instead of just throwing an internal error and stopping judging.

### PR DESCRIPTION
**Pull request by meisterT**
This behavior can be disabled by starting the judgedaemon with
`--diskspace_error`.
If that flag is passed (or not enough diskspace is freed up), an
internal error is posted.

Judging directories are cleaned up in order of their last modification.

Fixes #943.

image

### CODE REVIEW

File: judge/judgedaemon.main.php [hunk 1]
```diff
@@ -438,6 +439,7 @@ function fetch_executable(
 if (isset($options['n'])) {
     $options['daemonid'] = $options['n'];
 }
+if (isset($options))
```
Comment (nickygerritsen) [inline judge/judgedaemon.main.php:442]: You miss something here

Comment (meisterT) [reply judge/judgedaemon.main.php]: Done.

**Review by nickygerritsen — APPROVED** [review]

File: judge/judgedaemon.main.php [hunk 2]
```diff
@@ -612,29 +613,63 @@ function fetch_executable(
...
+                    return filemtime($a) <=> filemtime($b);
+                });
+                $after = $before = disk_free_space(JUDGEDIR);
+                logmsg(LOG_INFO,
+                    "🗑 Low on diskspace, cleaning up (" . count($candidateDirs) . " potential candidates).");
+                $cnt = 0;
+                foreach ($candidateDirs as $d) {
+                    $cnt++;
+                    logmsg(LOG_INFO, "  - deleting $d");
+                    system('rm -rf ' . dj_escapeshellarg("$d"), $retval);
```
Comment (eldering) [inline judge/judgedaemon.main.php:640]: No need to write `$d` inside an interpolated string.

**Review by eldering — APPROVED** [review]
LGTM except for a nitpick

**Comment by thijskh** [comment]
Is it just me who thinks the _ should be a - in the commandline parameter name?

**Comment by meisterT** [comment]
Thijs, no opinion here, so feel free to change.

**Comment by eldering** [comment]
> Is it just me who thinks the _ should be a - in the commandline parameter name?

I'd also have a slight preference for `-`.

**Pull request merged**